App/Sagas/NotificationsSagas.ts

```diff
@@ -81,10 +94,34 @@ export function * refreshNotifications () {
     const busy = yield select(NotificationsSelectors.refreshing)
     // skip multi-request back to back
     if (busy) return
+    yield * waitUntilOnline(1000)
     yield put(NotificationsActions.refreshNotificationsStart())
     const notificationResponse = yield call(TextileNode.getNotifications, 99)
     yield put(NotificationsActions.refreshNotificationsSuccess(notificationResponse.items))
   } catch (error) {
     yield put(NotificationsActions.refreshNotificationsFailure())
   }
 }
+
+export function * reviewThreadInvite (action: ActionType<typeof NotificationsActions.reviewNotificationThreadInvite>) {
+  const {notification} = action.payload
+  try {
+    const payload = NotificationsServices.toPayload(notification)
+    if (!payload) return
+    yield call(NotificationsServices.displayInviteAlert, payload.message)
+    yield put(ThreadsActions.acceptInviteRequest(notification.id, notification.category))
+  } catch (error) {
+    // Ignore invite
+  }
+}
+
+function * waitUntilOnline(ms: number) {
+  let ttw = ms
+  let online = yield select(TextileNodeSelectors.online)
+  while(!online && 0 < ttw) {
+    yield delay(50)
+    online = yield select(TextileNodeSelectors.online)
+    ttw -= 50
+  }
+  return online
```

Comment:
This works, but just FYI a cool way to do this:
```
let online = yield select(TextileNodeSelectors.online)
if (!online) {
  yield race({
    online: take(getType(TextileNodeActions.online)),
    timeout: delay(ms)
  })
}
```

Reply:
where does getType come from?

Reply:
`typesafe-actions`. There are some uses of `take` with `getType` in `TextileSagas` already.